x/staking/keeper/delegation.go

```diff
@@ -172,26 +172,29 @@ func (k Keeper) GetUnbondingDelegation(ctx context.Context, delAddr sdk.AccAddre
 // particular validator.
 func (k Keeper) GetUnbondingDelegationsFromValidator(ctx context.Context, valAddr sdk.ValAddress) (ubds []types.UnbondingDelegation, err error) {
 	store := k.storeService.OpenKVStore(ctx)
-	prefix := types.GetUBDsByValIndexKey(valAddr)
-	iterator, err := store.Iterator(prefix, storetypes.PrefixEndBytes(prefix))
+	rng := collections.NewPrefixUntilPairRange[sdk.ValAddress, sdk.AccAddress](valAddr)
```

Comment:
this does not look correct, and we should add a test that asserts this behaviour (in case no test is failing here), prefixed until means, as the word says, that the iteration will prefix until the provided prefix. 

so assuming the following state:


| val1-del1 | ubd1 |
| val1-del2| ubd2 |
| val2-del1| ubd3 |

doing PrefixedUntil(val2) will also return results from val1. 

instead we should be using collections.NewPrefixedPairRange which will include in the results only the keys starting with the provided prefix.